src/status_im2/contexts/syncing/sheets/enter_password/views.cljs

```diff
@@ -0,0 +1,87 @@
+            [status-im2.contexts.syncing.sheets.enter-password.styles :as styles]
+            [re-frame.core :as re-frame]
+            [quo2.core :as quo2]
+            [quo.core :as quo]
+            [i18n.i18n :as i18n]
+            [status-im.ui.components.react :as react]
+            [status-im.ui.components.toolbar :as toolbar]
+            [clojure.string :as string]
+            [taoensso.timbre :as log]
+            [status-im.utils.security :as security]
+            [status-im.ethereum.core :as ethereum]
+            [status-im.utils.fx :as fx]
+            [status-im.native-module.core :as status]
+            [status-im.ui.components.qr-code-viewer.views :as qr-code-viewer]))
+
+(def entered-password (atom ""))
+(def connection-string  (atom ""))
+
+(defn qr-code-view-with-connection-string [connection-string]
+  (log/info "connection-string ===> " connection-string)
+  (let [window-width       @(re-frame/subscribe [:dimensions/window-width])
+        valid-cs?          (string/starts-with? connection-string "cs")]
+    [:<>
+     (if valid-cs?
+       [rn/view {:style styles/body-container}
+        [rn/text {:style styles/header-text} (i18n/label :t/sync-code-generated)]
+        [qr-code-viewer/qr-code-view (* window-width 0.808) connection-string]
+        [quo2/information-box {:type      :informative
+                               :closable? false
+                               :icon      :main-icons2/placeholder
+                               :style     {:margin-top 20}} (i18n/label :t/instruction-after-qr-generated)]]
+       [rn/view {:style styles/body-container}
+        [rn/text {:style styles/header-text} connection-string]
+        [toolbar/toolbar
+         {:size :large
+          :center
+          [react/view {:padding-horizontal 8}
+           [quo/button
+            {:on-press #(re-frame/dispatch [:preperations-for-connection-string])}
+            (i18n/label :t/try-your-luck-again)]]}]])]))
+
+(fx/defn initiate-local-pairing-with-connection-string
+  {:events [:initiate-local-pairing-with-connection-string]}
+  [{:keys [db]} {:keys [data]}]
+  (let [config-map         (.stringify js/JSON (clj->js {:keyUID "" :keystorePath "" :password ""}))
+        connection-string  data]
+    (status/input-connection-string-for-bootstrapping
+     connection-string
+     config-map
+     #(log/info "this is response from initiate-local-pairing-with-connection-string " %))))
+
+(fx/defn preperations-for-connection-string
+  {:events [:preperations-for-connection-string]}
+  [{:keys [db]}]
+  (let [sha3-pwd           (ethereum/sha3 (security/safe-unmask-data @entered-password))
+        key-uid            (get-in db [:multiaccount :key-uid])
+        config-map         (.stringify js/JSON (clj->js {:keyUID key-uid :keystorePath "" :password sha3-pwd}))]
+    (status/get-connection-string-for-bootstrapping-another-device
+     config-map
+     #(re-frame/dispatch [:bottom-sheet/show-sheet
```

Comment:
The function reader macro `#` is a tad confusing here. It should be left for one liners or at least avoided in cases where the dev has to scan multiple lines for the magical `%` sign. Here I would suggest the following:

```clojure
(fx/defn preperations-for-connection-string
  {:events [:preperations-for-connection-string]}
  [{:keys [db]}]
  (let [sha3-pwd           (ethereum/sha3 (security/safe-unmask-data @entered-password))
        key-uid            (get-in db [:multiaccount :key-uid])
        config-map         (.stringify js/JSON (clj->js {:keyUID key-uid :keystorePath "" :password sha3-pwd}))]
    (status/get-connection-string-for-bootstrapping-another-device
     config-map
     (fn [connection-string]
       (re-frame/dispatch [:bottom-sheet/show-sheet
                           {:showHandle? false
                            :content (fn []
                                       [qr-code-view-with-connection-string connection-string])}])))))
```